aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -342,376 +365,602 @@ void upsample_avx_bilinear_uint8(
+          _mm_loadl_epi64((__m128i*)&lineIn0_min[i])),
+          _mm_loadl_epi64((__m128i*)&lineIn1_min[i]), 1);
+
+      // Apply mask_low:
+      //   [r0 g0 b0 a0  r1 g1 b1 a1  ... | R0 G0 B0 A0  R1 G1 B1 A1 ... ] ->
+      //   [r0 0 r1 0  g0 0 g1 0  b0 0 b1 0  a0 0 a1 0 | R0 0 R1 0  G0 0 G1 0  B0 0 B1 0  A0 0 A1 0]
+      auto pix1 = _mm256_shuffle_epi8(source1, mask_low);
+      // Compute output value as C += w0 * C0 + w1 * C1 for each channel in 32-bit precision
+      sss0 = _mm256_add_epi32(sss0, _mm256_madd_epi16(pix1, mmk));
+
+      // Same as above for lines 2 and 3:
+      auto source2 = _mm256_inserti128_si256(_mm256_castsi128_si256(
+          _mm_loadl_epi64((__m128i*)&lineIn2_min[i])),
+          _mm_loadl_epi64((__m128i*)&lineIn3_min[i]), 1);
+      auto pix2 = _mm256_shuffle_epi8(source2, mask_low);
+      sss1 = _mm256_add_epi32(sss1, _mm256_madd_epi16(pix2, mmk));
     }
 
-    for (; x < xmax; x++) {
-      __m256i pix, mmk;
-
-      // [16] xx k0 xx k0 xx k0 xx k0 xx k0 xx k0 xx k0 xx k0
-      mmk = _mm256_set1_epi32(k[x]);
-
-      // [16] xx a0 xx b0 xx g0 xx r0 xx a0 xx b0 xx g0 xx r0
-      pix = _mm256_inserti128_si256(
-          _mm256_castsi128_si256(mm_cvtepu8_epi32(&lineIn0[x + xmin])),
-          mm_cvtepu8_epi32(&lineIn1[x + xmin]),
-          1);
-      sss0 = _mm256_add_epi32(sss0, _mm256_madd_epi16(pix, mmk));
-
-      pix = _mm256_inserti128_si256(
-          _mm256_castsi128_si256(mm_cvtepu8_epi32(&lineIn2[x + xmin])),
-          mm_cvtepu8_epi32(&lineIn3[x + xmin]),
-          1);
-      sss1 = _mm256_add_epi32(sss1, _mm256_madd_epi16(pix, mmk));
+    // block 1
+    for (; i < ids_size; i++) {
+      // Load 1 value from weight vector
+      // mmk = [wl_0 wh_0 0 0  wl_0 wh_0 0 0  ...]
+      const auto mmk = _mm256_set1_epi32(k[i]);
+
+      // Load 2 pixels (one per line) from input lines 0 and 1:
+      // source = [
+      //   r0 g0 b0 a0  0 0 0 0  0 0 0 0  0 0 0 0
+      //   R0 G0 B0 A0  0 0 0 0  0 0 0 0  0 0 0 0
+      // ]
+      auto pix1 = _mm256_inserti128_si256(_mm256_castsi128_si256(
+          mm_cvtepu8_epi32(&lineIn0_min[i])),
+          mm_cvtepu8_epi32(&lineIn1_min[i]), 1);
+      // Compute output value as C += w0 * C0 for each channel in 32-bit precision
+      sss0 = _mm256_add_epi32(sss0, _mm256_madd_epi16(pix1, mmk));
+
+      // Same as above for lines 2 and 3
+      auto pix2 = _mm256_inserti128_si256(_mm256_castsi128_si256(
+          mm_cvtepu8_epi32(&lineIn2_min[i])),
+          mm_cvtepu8_epi32(&lineIn3_min[i]), 1);
+      sss1 = _mm256_add_epi32(sss1, _mm256_madd_epi16(pix2, mmk));
     }
 
+    // Shift back the output integer values: output = output >> weights_precision
     sss0 = _mm256_srai_epi32(sss0, coefs_precision);
     sss1 = _mm256_srai_epi32(sss1, coefs_precision);
+    // Convert packed signed 32-bit integers to packed 16-bit integers using signed saturation
+    // (a a a a b b b b | c c c c d d d d) -> (a a b b c c d d | 0 0 0 0 0 0 0 0)
```

Comment:
Why depict the bar in the middle of the 128-bit lane, where before you have used it to separate complete lanes?